crates/red_knot_workspace/src/site_packages.rs

```diff
@@ -25,26 +77,92 @@ use ruff_db::system::{System, SystemPath, SystemPathBuf};
 /// will be `/opt/homebrew`, and `site-packages` will be at
 /// `/opt/homebrew/lib/python3.X/site-packages`.
 ///
-/// This routine does not verify that `sys_prefix_path` points
-/// to an existing directory on disk; it is assumed that this has already
-/// been checked.
-///
 /// [`sys.prefix`]: https://docs.python.org/3/library/sys.html#sys.prefix
+#[derive(Debug, PartialEq, Eq)]
+struct SysPrefixPath(SystemPathBuf);
+
+impl SysPrefixPath {
+    fn new(unvalidated_path: impl AsRef<SystemPath>, system: &dyn System) -> Option<Self> {
+        let unvalidated_path = unvalidated_path.as_ref();
+        let canonicalized = system
+            .canonicalize_path(unvalidated_path)
+            .unwrap_or_else(|_| unvalidated_path.to_path_buf());
```

Comment:
Same as above. Do we need an absolute path or do we need to resolve symlinks?

I feel like we should return an `Error` if `canonicalize` fails (if canonicalizing is important)